index.js

```diff
@@ -1 +1,438 @@
-module.exports = require('./lib/eql');
+'use strict';
+/* globals Symbol: true, Uint8Array: true, WeakMap: true */
+/*!
+ * deep-eql
+ * Copyright(c) 2013 Jake Luer <jake@alogicalparadox.com>
+ * MIT Licensed
+ */
+
+/*!
+ * Module dependencies
+ */
+
+var type = require('type-detect');
+var objectIs = Object.is || require('object-is'); // eslint-disable-line
+
+function FakeMap() {
+  this.clear();
+}
+FakeMap.prototype = {
+  clear: function clearMap() {
+    this.keys = [];
+    this.values = [];
+    return this;
+  },
+  set: function setMap(key, value) {
+    var index = this.keys.indexOf(key);
+    if (index >= 0) {
+      this.values[index] = value;
+    } else {
+      this.keys.push(key);
+      this.values.push(value);
+    }
+    return this;
+  },
+  get: function getMap(key) {
+    return this.values[this.keys.indexOf(key)];
+  },
+  delete: function deleteMap(key) {
+    var index = this.keys.indexOf(key);
+    if (index >= 0) {
+      this.values = this.values.slice(0, index).concat(this.values.slice(index + 1));
+      this.keys = this.keys.slice(0, index).concat(this.keys.slice(index + 1));
+    }
+    return this;
+  },
+};
+
+var MemoizeMap = null;
+if (typeof WeakMap === 'function') {
+  MemoizeMap = WeakMap;
+} else {
+  MemoizeMap = FakeMap;
+}
+
+function memoizeCompare(leftHandOperand, rightHandOperand, memoizeMap) {
+  var leftHandMap = memoizeMap.get(leftHandOperand);
+  if (leftHandMap) {
+    var result = leftHandMap.get(rightHandOperand);
+    if (typeof result === 'boolean') {
+      return result;
+    }
+  }
+  return null;
+}
+
+function memoizeSet(leftHandOperand, rightHandOperand, memoizeMap, result) {
+  if (!memoizeMap) {
+    return;
+  }
+  var leftHandMap = memoizeMap.get(leftHandOperand);
+  if (leftHandMap) {
+    leftHandMap.set(rightHandOperand, result);
+  } else {
+    leftHandMap = new MemoizeMap();
+    leftHandMap.set(rightHandOperand, result);
+    memoizeMap.set(leftHandOperand, leftHandMap);
+  }
+}
+
+/*!
+ * Primary Export
+ */
+
+module.exports = deepEqual;
+module.exports.MemoizeMap = MemoizeMap;
+
+/**
+ * Assert deeply nested sameValue equality between two objects of any type.
+ *
+ * @param {Mixed} leftHandOperand
+ * @param {Mixed} rightHandOperand
+ * @param {Array} [comparator] (optional) - Override default algo, determening custom equality
+ * @param {Array} [memoize] (optional) - Memoize the results of complex objects for a speed boost
+ * @return {Boolean} equal match
+ */
+
+// Fast comparisons go through this small function for optimisation
+function deepEqual(leftHandOperand, rightHandOperand, options) {
+  var leftHandType = typeof leftHandOperand;
+  if (
+    leftHandOperand === null ||
+    leftHandOperand === undefined || // eslint-disable-line no-undefined
+    leftHandType === 'boolean' ||
+    leftHandType === 'string'
+  ) {
```

Comment:
FYI our tests are currently run on Saucelabs OSX 10.11, which resolves to Safari 9.0 (check https://wiki.saucelabs.com/display/DOCS/Platform+Configurator#/). 

Also:

> One issue with this is that boolean primitives and their corresponding object wrappers should be equal. They are not in this case

Not sure about this one. They have no enumerable keys - and so we can only check for referential equality unless we have a specific code paths to deal with those objects. As you say, they are an anti pattern anyway - so is there any point stretching to support them?